apps/arena/lib/arena/game_updater.ex

```diff
@@ -485,6 +488,38 @@ defmodule Arena.GameUpdater do
     |> Map.put(:aditional_info, entity |> Entities.maybe_add_custom_info())
   end
 
+  defp report_game_results(state, winner_id) do
+    results =
+      Enum.filter(state.game_state.client_to_player_map, fn {client_id, _player_id} ->
+        Enum.member?(state.clients, client_id)
+      end)
+      |> Enum.map(fn {client_id, player_id} ->
+        player = Map.get(state.game_state.players, player_id)
+
+        %{
+          user_id: client_id,
+          ## TODO: way to track `abandon`, currently a bot taking over will endup with a result
+          result: if(player.id == winner_id, do: "win", else: "loss"),
+          kills: player.aditional_info.kill_count,
+          ## TODO: this only works because you can only die once
+          deaths: if(Player.alive?(player), do: 0, else: 1),
+          character: player.aditional_info.character_name
+        }
+      end)
+
+    payload = Jason.encode!(%{match_id: Ecto.UUID.generate(), results: results})
+
+    ## TODO: we should be doing this in a better way, both the url and the actual request
+    ## maybe a separate GenServer that gets the results and tries to send them to the server?
+    ## This way if it fails we can retry or something
+    spawn(fn ->
+      gateway_url = System.get_env("GATEWAY_URL") || "http://localhost:4001"
+
+      Finch.build(:post, "#{gateway_url}/arena/match", [{"content-type", "application/json"}], payload)
+      |> Finch.request(Arena.Finch)
```

Comment:
remember that Finch.request is sync and could cause the game_updater to timeout if the game_backend app isn't reachable, you can replace this with `async_request` or spawn a new process to handle this request, pretty similar to what `game_launcher` is doing

Reply:
This is inside a `spawn/1` so no issues with that

Reply:
oh right, i didn't saw it 😅